jira/client.py

```diff
@@ -1660,9 +1660,19 @@ def _get_user_id(self, user: str) -> str:
         try:
             user_obj: User
             if self._is_cloud:
-                user_obj = self.search_users(query=user, maxResults=1)[0]
+                users = self.search_users(query=user)
             else:
-                user_obj = self.search_users(user=user, maxResults=1)[0]
+                users = self.search_users(user=user)
+            
+            if len(users) > 1:
+                matches = [u for u in users if u.key == user]
+                if len(matches) >= 1:
+                    user_obj = matches[0]
+                else:
+                    user_obj = users[0]               
+            else:
+                user_obj = users[0]
```

Comment:
```suggestion
            user_obj = matches[0] if matches else users[0]
```

and `matches = []` before the `if len(users) > 1`